File: spendy_backend/src/main/java/com/rohitcodes/expense_tracker/service/AuthService.java
package com.rohitcodes.expense_tracker.service;

import com.rohitcodes.expense_tracker.dto.auth.AuthResponse;
import com.rohitcodes.expense_tracker.dto.auth.LoginRequest;
import com.rohitcodes.expense_tracker.dto.auth.RegisterRequest;
import org.springframework.context.annotation.Bean;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}

File: spendy-mobileFrontend/src/services/authService.ts
import { api } from "./api";
import type {
  AuthResponse,
  LoginRequest,
  RegisterRequest,
} from "@/src/types/auth";

export const authService = {
  async login(payload: LoginRequest) {
    const response = await api.post<AuthResponse>(
      "/auth/login",
      payload
    );
    return response.data;
  },

  async register(payload: RegisterRequest) {
    const response = await api.post<AuthResponse>(
      "/auth/register",
      payload
    );
    return response.data;
  },
};
